app/models/loader/types.rb

```diff
@@ -96,43 +96,58 @@ def create_resource!
           resource.annotations_dataset.import(%i(resource_id name value), records)
         end
       end
-      
+
       def resource
         ::Resource[resourceid]
       end
     end
-      
+
     class Record < Types::Base
       include CreateRole
       include CreateResource
-      
+
+      def verify
+        message = "Verify method for entity #{self} does not exist"
+        raise Exceptions::InvalidPolicyObject.new(self.id, message: message)
+      end
+
+      def calculate_defaults!; end
```

Comment:
Unlike in Java, where you explicitly define interfaces, we should never have an empty method like this in ruby.  Ruby uses duck-typing instead.

Reply:
@jonahx thank you for your comments. I agree that empty methods is not the perfect way - but if I would not use them I could not have a method in Record class (verify method which throws exception). And it is essential to enforce people to implement verification. I also plan that all the verify methods that are currently empty will be filled soon by applicational logic (relevant verifications). I just could not do it here because it would change customer-facing functionality